src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -254,10 +254,10 @@ private void SetCultureOnUnstartedThreadNoCheck(CultureInfo value, bool uiCultur
             }
         }
 
-        internal ExecutionContext ExecutionContext
+        public ExecutionContext ExecutionContext
```

Comment:
Does this internal ExecutionContext state have the same behavior as the public ExecutionContext property?

E.g. When this returns null vs. when the public Thread.ExecutionContext returns null.

I think we may need to change CoreLib to use a different getter internally; and change the public getter to call `ExecutionContext.Capture()` like what it used to do before.

Reply:
> Does this internal ExecutionContext state have the same behavior as the public ExecutionContext property?

Probably not, but I wanted to fix the surface area first.

Reply:
I would expect tests to fail if this is not done right. (If there are no tests failing, then we have a test hole.)

Reply:
Yup.  I'm just working through the failures (interspersed with meetings :)).